lib/libvfio-user.c

```diff
@@ -718,7 +726,8 @@ handle_dma_map(vfu_ctx_t *vfu_ctx, vfu_msg_t *msg,
         }
     }
 
-    ret = dma_controller_add_region(vfu_ctx->dma, (void *)dma_map->addr,
+    ret = dma_controller_add_region(vfu_ctx->dma,
+                                    (vfu_dma_addr_t)(uintptr_t)dma_map->addr,
```

Comment:
Perhaps we should use `uinptr_t` instead of `void *` for `vfu_dma_addr_t`?

Reply:
I think uintptr_t is better yeah

Reply:
We'll need a lot of casts because we compare `struct iovec.iov_base` to `vfu_dma_addr_t`, not sure it's worth it. Unless we drop `struct iovec.iov_base`, which I'm not sure it's nice either.